parsers/number.go

```diff
@@ -35,21 +37,17 @@ func (n *NumberGroup) CanParseIntoHuman(s string) (bool, error) {
 // 	Should at least be the number 1 thousand
 // 	Can't have letters in it
 // 	Needs to have a comma in it
-func (n *NumberGroup) CanParseFromHuman(s string) bool {
-
-	if len(s) <= 4 {
-		return false
-	}
+func (n *NumberGroup) CanParseFromHuman(s string) (bool, error) {
 
-	if match, _ := regexp.MatchString(`[a-z]+`, s); match {
-		return false
-	}
-
-	if !strings.Contains(s, ",") {
-		return false
+	if isMachineNumber(s) && len(s) < 4 {
+		return false, ErrTooSmall
+	} else if isMachineNumber(s) {
+		return false, ErrNotHumanGroup
+	} else if isDelimitedNumber(s) {
+		return true, nil
+	} else {
```

Comment:
since this one is not like the others in that it's the _wanted_ branch and not an error classification;  I would move this outside of this block.

```go
if isDelimtedNumber(s) {
    return true, nil}
}

// Error cases
var err error
if isMachineNumber() {
    err = ErrToSmall
else if ... {
   err = ErrFoobar
}
return false, errr
```

This way we can early exit and denote that the rest of the body is just classifying the _why_ we can't parse